File: F#/SingleLinked.fs
module HowProgrammingWorks.FS.SingleLinked

open System

type Node<'T> = {
    mutable next : Node<'T> option;
    mutable data : 'T;
}

let empty() = {next = None; data = None; }

let pust node data =
    Some { next = node; data = data }


let Example =

    let rec write (x : Node<string> option)  =
        match x with
            | None -> 
                Console.WriteLine()
            | Some n -> 
                printf "%s " n.data
                write n.next

    let n6 : Node<string> = {next = None; data = "road";}
    let n5 : Node<string> = {next = Some n6; data = "the";}
    let n4 : Node<string> = {next = Some n5; data = "across";}
    let n3 : Node<string> = {next = Some n4; data = "running";}
    let n2 : Node<string>= {next = Some n3; data = "was";}
    let n1 : Node<string>= {next = Some n2; data = "cat";}
    let n0 : Node<string>= {next = Some n1; data = "Black";}

    write (Some n0)

File: C#/SingleLinked.cs
using System;

namespace HowProgrammingWorks.CS.SingleLinkedList
{
    class Node<T>
    {
        public Node(T data)
            :  this(null, data)
        {            
        }

        public Node(Node<T> next, T data)
        {
            Next = next;
            Data = data;
        }

        public Node()
            : this(null, default(T))
        {
        }

        public Node<T> Next { get; set; }
        public T Data { get; set; }
    }

    class Example
    {
        public void Run()
        {
            var n6 = new Node<string>("road");
            var n5 = new Node<string>(n6, "the");
            var n4 = new Node<string>(n5, "across");
            var n3 = new Node<string>(n4, "running");
            var n2 = new Node<string>(n3, "was");
            var n1 = new Node<string>(n2, "cat");
            var n0 = new Node<string>(n1, "Black");

            var currentNode = n0;
            
            while (currentNode != null)
            {
                Console.Write($"{currentNode.Data} ");
                currentNode = currentNode.Next;
            }

            Console.WriteLine();
        }
    }
}
